pkg/plugins/golang/v4/scaffolds/internal/templates/cmd/featuregates.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"sigs.k8s.io/kubebuilder/v4/pkg/machinery"
+)
+
+var _ machinery.Template = &FeatureGates{}
+
+// FeatureGates scaffolds a feature gates package
+type FeatureGates struct {
+	machinery.TemplateMixin
+	machinery.BoilerplateMixin
+	machinery.RepositoryMixin
+
+	// Available feature gates discovered from the project
+	AvailableGates []string
+}
+
+// SetTemplateDefaults implements machinery.Template
+func (f *FeatureGates) SetTemplateDefaults() error {
+	if f.Path == "" {
+		f.Path = "internal/featuregates/featuregates.go"
+	}
+
+	f.TemplateBody = featureGatesTemplate
+
+	return nil
+}
+
+const featureGatesTemplate = `{{ .Boilerplate }}
+
+package featuregates
```

Comment:
How is your idea?
Could we have a UX where we minimise the utilities in the scaffolds?